File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Java/Exercises/37_compito_java_2020_01_24/Socio.java
import java.util.*;

public class Socio {
    protected int codice;
    protected String nomeCognome;
    protected int eta, categoria;
    protected List<Prenotazione> prenotazioni;

    public Socio(int codice, String nomeCognome, int eta, int categoria) {
        this.codice = codice;
        this.nomeCognome = nomeCognome;
        this.eta = eta;
        this.categoria = categoria;
        prenotazioni = new LinkedList<Prenotazione>();
    }

    public void addPrenotazione(Prenotazione p) {
        prenotazioni.add(p);
    }

    public String toString() {
        return codice + "\t" + nomeCognome + "\t" + eta + "\t" + categoria + "\t" + prenotazioni;
    }
}

File: Programming-languages-and-laboratory/Python/Lessons/26_20221202/37_soluzione/Socio.py
class Socio:
    def __init__(self, codice, nome_cognome, eta, categoria):
        self._codice = codice
        self._nome_cognome = nome_cognome
        self._eta=eta
        self._categoria=categoria
        self._prenotazioni = []

    def addPrenotazione(self, pren):
        self._prenotazioni.append(pren)


    def __str__(self):
        return str(self._codice)+"\t"+self._nome_cognome + "\t"  + str(self._eta)+"\t"+str(self._categoria)+"\t"+str([str(pren) for pren in self._prenotazioni])
